### ISSUE TITLE
Rule for infinite loop detection

### ISSUE THREAD

**Issue by sonia1588** [palantir/tslint#3693]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.5.3
- __Running TSLint via__:  VSCode 

#### TypeScript code being linted

```ts
export class WelcomeComponent implements OnInit {
   private a: any = 2;
   private b: any = 11;
  constructor() {}

ngOnInit() {
    
for (let i = this.a; i < this.b; i++) {
        console.log( 'Hello' + i);
        this.b++;
    }

```

#### Expected behavior
in above code snippet, for loop results infinite loop, Is there any rule which  tells the developer about infinite loop

**Comment by giladgray** [comment]
@sonia1588 unfortunately no, and unlikely to have such a solution for several decades, if ever.
https://en.wikipedia.org/wiki/Halting_problem

**Comment by sonia1588** [comment]
@giladgray , Thank you for quick response.
